LinearRegression/Blog/main.tex

```diff
@@ -0,0 +1,235 @@
+\begin{figure}[H]\centering\includegraphics[width=\linewidth]{23}\end{figure}
+\begin{figure}[H]\centering\includegraphics[width=1.05\linewidth]{24}\end{figure}
+Note that apart from the mean and standard error, summary\_frame() spits out two intervals at 95\% confidence level (we have set alpha = 0.05). The first pair of lower bound and upper bound forms a confidence interval, which is an estimate of the mean price of all the houses that match our randomly generated data. The second pair, on the other hand, forms a prediction interval, which estimate the price of a particular house with this set of data. Because individual values vary more than means, we can expect the prediction interval to be wider than the confidence interval at the same confidence level.
+
+\section*{Evaluations}
+While it is relatively straightforward to obtain regression lines and predictions, there are numerous evaluation metrics to measure the accuracy of our model. In fact, the summary() function from statsmodels provides you with an overwhelming amount of data:
+\begin{figure}[H]\includegraphics[width=\linewidth]{25}\end{figure}
+Don't panic, let's interpret some important values one by one.
+\subsubsection*{R-squared and Adjusted R-squared}
+\begin{figure}[H]\includegraphics[width=0.5\linewidth]{26}\end{figure}
+R-squared, a.k.a. coefficient of determination, measures the proportion of the variance of the target variable that can be explained by the predictors in our model. It is calculated as $1 - \frac{\text{Sum of Squares Error (SSE)}}{\text{Sum of Squares Total (SST)}}$, and ranges from 0 to 100\%. The higher $R^2$, the better the model. An $R^2$ of 100\%, for example, means the model explains all the variation of the target variable, whereas a value of 0\% indicates zero predictive capability.
+Our model has $R^2 = 0.542$, so approximately 54\% of the variation in the house price can be accounted for by our OLS model.
+
+Note that there's an adjusted $R^2 = 0.537$ in the second row which is slightly smaller than $R^2$. This is because $R^2$ only works as intended in a simple linear regression model where there's only one predictor. In a multiple regression, when a new predictor is added to the model, $R^2$ can only increase but never decrease. This implies that a model may seem to have a better fit simply because it has more predictors. Thus the adjusted $R^2$, calculated as $1 - \frac{\text{Mean Squares Error (MSE)}}{\text{Mean Squares Total (MST)}}$ or $1 - \frac{(1 - R^2)(n - 1)}{n - k - 1}$ if we already know $R^2$, takes into account the number of predictors $k$ and the number of data points $n$ in the model. It only increases if the new predictor improves the model more than expected by chance and decreases if it fails to do so. As a result, adjusted $R^2$ should always be less than or equal to $R^2$.
```

Comment:
This paragraph has very detailed explanations, but if possible, see which sentences could be shortened- this will make the paragraph more concise!